test/common/http/http1/codec_impl_test.cc

```diff
@@ -2826,5 +2886,29 @@ TEST_P(Http1ClientConnectionImplTest, ManyResponseHeadersAccepted) {
   status = codec_->dispatch(buffer);
 }
 
+TEST_P(Http1ServerConnectionImplTest, TestSmugglingDisallowChunkedContentLength0) {
+  testAllowChunkedContentLength(0, false);
+}
+TEST_P(Http1ServerConnectionImplTest, TestSmugglingDisallowChunkedContentLength1) {
+  // content-length less than POST body size
+  testAllowChunkedContentLength(1, false);
+}
+TEST_P(Http1ServerConnectionImplTest, TestSmugglingDisallowChunkedContentLength100) {
+  // content-length greater than POST body size
+  testAllowChunkedContentLength(100, false);
+}
+
+TEST_P(Http1ServerConnectionImplTest, TestSmugglingAllowChunkedContentLength0) {
+  testAllowChunkedContentLength(0, true);
+}
+TEST_P(Http1ServerConnectionImplTest, TestSmugglingAllowChunkedContentLength1) {
+  // content-length less than POST body size
+  testAllowChunkedContentLength(1, true);
+}
+TEST_P(Http1ServerConnectionImplTest, TestSmugglingAllowChunkedContentLength100) {
+  // content-length greater than POST body size
+  testAllowChunkedContentLength(100, true);
+}
```

Comment:
I'm not seeing similar tests for the response side which operate on Http1ClientConnectionImpl.

Reply:
Found an issue with http parser in response test case, will updated PR when it's fixed.
/wait

Reply:
added tests for response case